File: apps/hub/src/api/portfolio.ts
import client from './client';

export interface Portfolio {
  id: string; slug: string; headline?: string; bio?: string;
  is_public: boolean; linkedin_url?: string; website_url?: string;
  user_name?: string; certificate_count: number;
}

export const getMyPortfolio = () => client.get<Portfolio>('/v1/portfolio/me');
export const updateMyPortfolio = (data: Partial<Portfolio>) => client.patch<Portfolio>('/v1/portfolio/me', data);
export const getPublicPortfolio = (slug: string) => client.get<Portfolio>(`/v1/portfolio/${slug}`);

File: apps/api/app/db/models/portfolio.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID

from app.db.base import Base


class Portfolio(Base):
    """User's public portfolio page."""
    __tablename__ = "portfolios"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False, unique=True)
    slug = Column(String(100), nullable=False, unique=True)
    headline = Column(String(255), nullable=True)
    bio = Column(Text, nullable=True)
    is_public = Column(Boolean, default=True)
    linkedin_url = Column(Text, nullable=True)
    website_url = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
